File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        right=len(height)-1
        left=0
        maxwater=0
        while left<right:
            width=right-left
            minheight=min(height[left],height[right])
            area=width*minheight
            maxwater=max(maxwater,area)
            
            if height[left]<height[right]:
                left+=1
            else:
                right-=1
        
        return maxwater
        

        """
        :type height: List[int]
        :rtype: int
        """
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int right=height.size()-1,area;
        int left=0;
        int water=0;
        while(left<right){
            area=min(height[left],height[right]) * (right-left);

            water=max(area,water);

            if(height[left]<height[right]) left++;

            else right--;
        }

        return water;
    }
};